### PR TITLE
Fix install-range constraint.

### PR DESCRIPTION
**Pull request by floitsch**
Just using the given term as prefix doesn't work. For example, 'pkg install foo@1' would match 'v10.0.1'.

Also some refactorings to use conventions that are more in line with the rest of the repository.

### CODE REVIEW

File: tools/pkg/constraints.toit [hunk 1]
```diff
@@ -15,29 +15,60 @@
...
       else if constraint.prefix == "":
         constraints.add (SimpleConstraint "=" version)
       else:
         constraints.add (SimpleConstraint constraint.prefix version)
+    return Constraint --simple-constraints=constraints --source=source
+
+  static parse-range source/string -> Constraint:
+    parser := SemanticVersionParser source --allow-missing-minor
+    parse-result := parser.semantic-version --consume-all
+    // triple/TripleParseResult
```
Comment (kasperl) [inline tools/pkg/constraints.toit:52]: What are these comments?

File: tools/pkg/constraints.toit [hunk 2]
```diff
@@ -15,29 +15,60 @@
...
+    // pre-releases/List
+    // build-numbers/List
+    triple := parse-result.triple.triple
```
Comment (kasperl) [inline tools/pkg/constraints.toit:55]: This looks kinda silly. Why not have major, minor, and patch on the TripleParseResult class?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/pkg/constraints.toit (hunk 1 above)
Comment (floitsch) [reply tools/pkg/constraints.toit]: Removed the comments.
done.

File: tools/pkg/constraints.toit (hunk 2 above)
Comment (floitsch) [reply tools/pkg/constraints.toit:52]: I'm not super happy with the structure either.
Will put it on my list of refactoring TODOs.

**Pull request merged**